src/transformers/modeling_utils.py

```diff
@@ -3019,22 +3019,15 @@ def _fix_key(key):
         unexpected_keys = list(set(loaded_keys) - set(expected_keys))
 
         if is_accelerate_available():
+            model.tie_weights()
```

Comment:
@sgugger This line causes a test of `marian` failing due to garbage output values.

It was removed by @SunMarc in #24200 when I reported the same issue to him.

If this line is necessary, we might need some fix/change for marian or its test.

```
tests/models/marian/test_modeling_marian.py::TestMarian_FI_EN_V2::test_batch_generation_en_fr

(line 433)  AssertionError: Lists differ: ['I like to read books', 'I like watching football'] != ['obliterat obliterat obliterat obliterat o[1345 chars]ɰɰɰ']
```